### PR TITLE
Canvas: Add toggle for connection points

### LINKED ISSUE #61440: Canvas: Add toggle for connection anchor UI
Add toggle in canvas panel options to enable / disable connection overlay (default to on?)

### PR DESCRIPTION
**Pull request by adela-almasan**
https://user-images.githubusercontent.com/88068998/213338633-5651e7fe-3f1d-4cc0-896a-672802042af2.mov

Fixes #61440

### CODE REVIEW

**Comment by ryantxu** [comment]
why would we want to disable connections?  all similar style tools have them enabled all the time (while editing)

File: public/app/plugins/panel/canvas/Connections.tsx [hunk 1]
```diff
@@ -192,7 +192,9 @@ export class Connections {
   render() {
     return (
       <>
-        <ConnectionAnchors setRef={this.setConnectionAnchorRef} handleMouseLeave={this.handleMouseLeave} />
+        {this.scene.shouldEnableConnections && (
```
Comment (ryantxu) [inline public/app/plugins/panel/canvas/Connections.tsx:195]: Any reason we need more control than this:
```suggestion
        {this.scene.isEditingEnabled && (
```

Comment (adela-almasan) [reply public/app/plugins/panel/canvas/Connections.tsx:195]: Currently they are enabled only while editing. Maybe we should've talked more about this, but I think this is a cleaner ui if you don't want to deal with connections. I believe draw.io has a checkbox to enable/disable them.

Comment (ryantxu) [reply public/app/plugins/panel/canvas/Connections.tsx:195]: oh I see, they do.  I am reluctant to add too many options while we still figure out what is what.

Comment (adela-almasan) [reply public/app/plugins/panel/canvas/Connections.tsx:195]: Fair point. We can put this on hold until we figure out if we need it or not.

**Pull request closed without merging**